protocol-units/settlement/mcr/contracts/src/staking/MovementStaking.sol

```diff
@@ -63,76 +49,45 @@ contract MovementStaking is
 
         // roll over from 0 (genesis) to current epoch by block time
         currentEpochByDomain[domain] = getEpochByBlockTime(domain);
-
-        for (uint256 i = 0; i < attestersByDomain[domain].length(); i++) {
-            address attester = attestersByDomain[domain].at(i);
-
-            for (uint256 j = 0; j < custodiansByDomain[domain].length(); j++) {
-                address custodian = custodiansByDomain[domain].at(j);
-
-                // get the genesis stake for the attester
-                uint256 attesterStake = getStakeAtEpoch(
-                    domain,
-                    0,
-                    custodian,
-                    attester
-                );
-
-                // roll over the genesis stake to the current epoch
-                _addStake(
-                    domain,
-                    getCurrentEpoch(domain),
-                    custodian,
-                    attester,
-                    attesterStake
-                );
+        EnumerableSet.AddressSet storage attesters = attestersByDomain[domain];
+        // for every custodian
+        EnumerableSet.AddressSet storage custodians = custodiansByDomain[domain];
+        for (uint256 i = 0; i < attesters.length(); i++) {
+            address attester = attesters.at(i);
+            // for every delegatee of the attester
+            EnumerableSet.AddressSet storage delegatees = delegatorsByAttesterByDomain[domain][attester];
+            for (uint256 j = 0; j < custodians.length(); j++) {
+                address custodian = custodians.at(j);
+
+                for (uint256 k = 0; k < delegatees.length(); k++) {
+                    // todo: can this be replaced with _rollOverAttester?
+                    address delegatee = delegatees.at(k);
+
```

Comment:
to use _rollOverAttester, it needs to start using nextEpoch param instead of currentEpoch. This way 0 can be passed as the param. This requires a modification of the rest of the code, but definitely seems okay.
Just rolling with the current implementation basicTest passes, it requires a test to verify if the correct epoch has been settled.